tests/avify4mtest.c

```diff
@@ -0,0 +1,177 @@
+// Copyright 2022 Google LLC. All rights reserved.
+// SPDX-License-Identifier: BSD-2-Clause
+
+#include "apps/shared/y4m.h"
+#include "avif/avif.h"
+
+#include <inttypes.h>
+#include <stdio.h>
+#include <stdlib.h>
+#include <string.h>
+
+static const char kPath[] = "/tmp/avify4mtest.y4m";
+
+//------------------------------------------------------------------------------
+
+// Returns true if image1 and image2 are identical.
+static avifBool compareYUVA(const avifImage * image1, const avifImage * image2)
+{
+    if (image1 == NULL || image2 == NULL || image1->width != image2->width || image1->height != image2->height ||
```

Comment:
Since `compareYUVA()` is an internal function, we can safely assume the callers never pass null pointers to this function, so we don't need to `check image1 == NULL || image2 == NULL`.